yash-env/src/system/real.rs

```diff
@@ -421,6 +422,36 @@ impl System for RealSystem {
     fn getpwnam_dir(&self, name: &str) -> nix::Result<Option<std::path::PathBuf>> {
         nix::unistd::User::from_name(name).map(|o| o.map(|passwd| passwd.dir))
     }
+
+    fn confstr_path(&self) -> nix::Result<OsString> {
+        // TODO Support other platforms
+        #[cfg(any(
+            target_os = "macos",
+            target_os = "ios",
+            target_os = "tvos",
+            target_os = "watchos"
+        ))]
+        unsafe {
+            use std::os::unix::ffi::OsStringExt as _;
+            let size = nix::libc::confstr(nix::libc::_CS_PATH, std::ptr::null_mut(), 0);
+            if size == 0 {
+                return Err(Errno::last());
+            }
+            let mut buffer = Vec::with_capacity(size);
+            let final_size = nix::libc::confstr(nix::libc::_CS_PATH, buffer.as_mut_ptr(), size);
+            if final_size == 0 {
+                return Err(Errno::last());
+            }
+            if final_size > size {
+                return Err(Errno::ERANGE);
+            }
+            buffer.set_len(final_size - 1); // The last byte is a null terminator.
+            Ok(OsString::from_vec(buffer))
+        }
+
+        #[allow(unreachable_code)]
+        Err(Errno::ENOSYS)
+    }
```

Comment:
The `nix::libc::confstr` function is only available on Apple OSes now, so I leave it as a to-do. I need to contribute to the libc crate so that `confstr` can be used on other platforms.